moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -339,6 +339,26 @@ bool planning_pipeline::PlanningPipeline::generatePlan(const planning_scene::Pla
     display_path_publisher_.publish(disp);
   }
 
+  if (!solved)
+  {
+    // This should alert the user if planning failed because of contradicting constraints.
+    // Could be checked more thoroughly, but it is probably not worth going to that length.
+    bool stacked_constraints = false;
+    if (req.goal_constraints.size() > 1 || req.path_constraints.joint_constraints.size() > 1 ||
+        req.path_constraints.position_constraints.size() > 1 || req.path_constraints.orientation_constraints.size() > 1)
+      stacked_constraints = true;
+    for (auto constraint : req.goal_constraints)
+    {
+      if (constraint.joint_constraints.size() > 1 || constraint.position_constraints.size() > 1 ||
```

Comment:
When setting joint state targets it's perfectly normal to have multiple joint constraints (1 for each joint) so the warning is not accurate. I would remove joint constraints from this entirely since they can't really contradict each other (except there are multiple for the same joint). Having multiple position or orientation constraints does not mean that multiple arms are constrained. The links can be and most often are part of the same arm and the goal pose is sampled to meet all such constraints. I think without an actual check if there are unfeasible constraints (i.e. directly conflicting constraints that apply to the same joint/link) this kind of warning doesn't really make sense.

Reply:
I had never used joint constraints and didn't notice each joint would be its own constraint. I agree and changed it.

Multiple position and orientation constraints seems to be an advanced use case to me. I feel like the most common case (and the only one that I remember seeing in the tutorials) is setting a 6D pose target for the end effector link through the move_group_interface, which results in one position and orientation constraint (each of them constraining 3 translations and rotations; I had to [double check](http://docs.ros.org/melodic/api/moveit_msgs/html/msg/Constraints.html)).

Like it says in the comment, a more thorough check seems nice, but nothing trivial comes to mind for me. I mean, you could check if the kinematic chain is overconstrained or multiple joint constraints apply to the same joint, but I don't know if it is worth the effort to avoid the false positives. This is more of a sanity check/nudge after a failed planning attempt to help users troubleshoot, which I assume advanced users would disregard if it doesn't apply to them. I think it's worth displaying this sort of nudge in general, because it can save a lot of user frustration, but I am not sure how the others feel about this.